### PR TITLE
Fix skipped tasks being collated in ListView's TaskBoxes

### PR DESCRIPTION
**Pull request by CaseyHillers**
A bug that started in https://github.com/flutter/cocoon/pull/518 where tasks that should be skipped or collated. This lead to skipped tasks not showing correctly and the grid becoming messed up.

The cause was becaused `SizedBox()` was not given a size for skipped tasks. With `GridView` this was okay as everything was constrained, however, this lead to them becoming hidden when switching to the `ListView` implementation.

## Tested

The only way I could think of regression testing this would be tapping at certain points on the screen and checking the task overlay to be correct. However, I did not know if this would become more so E2E testing of the application which is out of scope of the project.

Demo: https://testskipfix-dot-flutter-dashboard.appspot.com/v2/

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Consider placing a value key on the empty sized box - then you could would write a test that rendered the screen with missing tasks and searched for the valuekey + type, making an assertion on the size

**Comment by CaseyHillers** [comment]
Thanks for the suggestion! I was thinking at the bigger picture level (maybe because I noticed the bug by comparing the big TV version to my monitor :P ).

I added the ability to insert keys to the `StatusGrid` cells based on their location in the grid. This makes it easier to lookup the widget corresponding to a cell for comparison. Enhanced one of the previous tests to ensure skipped tasks existed, and followed up with a regression test to ensure all the cells in the grid have the same size.

File: app_flutter/test/status_grid_test.dart [hunk 1]
```diff
@@ -96,48 +96,50 @@ void main() {
...
+          ])),
+      CommitStatus()
+        ..stages.add(Stage()
+          ..name = 'A'
+          ..tasks.addAll(<Task>[
+            Task()
+              ..name = '3'
+              ..status = TaskBox.statusSucceeded
+          ]))
+    ];
```
Comment (digiter) [inline app_flutter/test/status_grid_test.dart]: I have difficulty in understanding how this commit status list be converted to the matrix diagram above.

Comment (CaseyHillers) [reply app_flutter/test/status_grid_test.dart]: Would it be helpful if I added the comment below the diagram stating:
```
/// To construct the matrix from this diagram, each [CommitStatus] must have a unique [Task] that does not share its name with any other [Task].
```

**Comment by CaseyHillers** [comment]
I accidentally overwrote this test version with a demo for downloading logs this morning. The new demo link is located at https://testskipfix-dot-flutter-dashboard.appspot.com/v2/

Updated demo link at top to reflect that.

File: app_flutter/test/status_grid_test.dart (hunk 1 above)
Comment (digiter) [reply app_flutter/test/status_grid_test.dart]: Yes, that would be helpful.

Comment (CaseyHillers) [reply app_flutter/test/status_grid_test.dart]: Done.

**Comment by CaseyHillers** [comment]
Uh oh, the tests got removed when I did a rebase. Let me fix that.

**Comment by CaseyHillers** [comment]
Tests readded. Must have been I forgot to fetch origin and didn't think twice on doing a force push.

File: app_flutter/test/status_grid_test.dart [hunk 2]
```diff
@@ -148,15 +153,78 @@ void main() {
...
+      expect(skippedTaskRow3Col2, isNotNull);
+
+      final TaskBox lastTask =
+          find.byKey(const Key('cell-2-2')).evaluate().first.widget;
+      expect(lastTask.task, statusesWithSkips[2].stages[0].tasks[0]);
+    });
+
+    testWidgets('all cells in the grid have the same size',
+        (WidgetTester tester) async {
+      /// Use [statusesWithSkips] to check edge cases with skipped tasks.
```
Comment (digiter) [inline app_flutter/test/status_grid_test.dart]: This comment is simply repeating the code. Consider removing it or merging it into the test description.

File: app_flutter/test/status_grid_test.dart [hunk 3]
```diff
@@ -148,15 +153,78 @@ void main() {
...
+          ),
+        ),
+      );
+
+      // Compare all the cells to the first cell to check they all have
+      // the same size
+      final Element taskBox =
+          find.byKey(const Key('cell-0-0')).evaluate().first;
+      for (int rowIndex = 0; rowIndex < statusesWithSkips.length; rowIndex++) {
+        for (int colIndex = 0; colIndex < 3; colIndex++) {
```
Comment (digiter) [inline app_flutter/test/status_grid_test.dart]: Avoid hard coded 3?

File: app_flutter/test/status_grid_test.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_flutter/test/status_grid_test.dart]: Thanks for pointing that out! I updated the test description to reflect the edge case the comment mentioned.

File: app_flutter/test/status_grid_test.dart (hunk 3 above)
Comment (CaseyHillers) [reply app_flutter/test/status_grid_test.dart]: Good catch! Switched these to use the `TaskMatrix` values.
